sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3188,21 +3201,39 @@ object GpuOverrides {
     exec[CustomShuffleReaderExec](
       "A wrapper of shuffle query stage",
       ExecChecks((TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_64 + TypeSig.ARRAY +
-        TypeSig.STRUCT + TypeSig.MAP).nested(), TypeSig.all),
+          TypeSig.STRUCT + TypeSig.MAP).nested(), TypeSig.all),
       (exec, conf, p, r) =>
-      new SparkPlanMeta[CustomShuffleReaderExec](exec, conf, p, r) {
-        override def tagPlanForGpu(): Unit = {
-          if (!exec.child.supportsColumnar) {
-            willNotWorkOnGpu(
-              "Unable to replace CustomShuffleReader due to child not being columnar")
+        new SparkPlanMeta[CustomShuffleReaderExec](exec, conf, p, r) {
+          override def tagPlanForGpu(): Unit = {
+            if (!exec.child.supportsColumnar) {
+              willNotWorkOnGpu(
+                "Unable to replace CustomShuffleReader due to child not being columnar")
+            }
+            val shuffleEx = exec.child.asInstanceOf[ShuffleQueryStageExec].plan
```

Comment:
How do we know that the child is a `ShuffleQueryStageExec`? That should only ever happen when AQE is enabled.  And even then I don't think it is a 100% guarantee.